src/components/organisms/AdminVenues/AdminVenues.tsx

```diff
@@ -0,0 +1,65 @@
+import React, { useMemo } from "react";
+import { Button } from "react-bootstrap";
+import { Link } from "react-router-dom";
+
+import { AnyVenue, VenueTemplate } from "types/venues";
+
+import { WithId } from "utils/id";
+
+import "./AdminVenues.scss";
+
+interface AdminVenuesProps {
+  venues: WithId<AnyVenue>[];
+}
+
+export const AdminVenues: React.FC<AdminVenuesProps> = ({ venues }) => {
+  const partyVenues = useMemo(() => {
+    return venues?.filter((venue) => venue.template === VenueTemplate.partymap);
+  }, [venues]);
+
+  const hasVenues: boolean = !!partyVenues.length;
+
+  return (
+    <div className="admin-venue">
+      <div className="admin-venue__create-button">
+        <h3>Admin Dashboard</h3>
+        <Button as={Link} to="/admin_v2/venue/creation">
+          Create a new space
+        </Button>
+      </div>
+      {!hasVenues && (
+        <div className="admin-venue__empty-venues">
+          <h3>Welcome!</h3>
+          <h3>Create your first Sparkle space</h3>
+        </div>
+      )}
+      {hasVenues && (
+        <div className="admin-venue__cards">
+          {partyVenues.map((venue, index) => {
+            return (
+              <div key={index} className="card">
+                <div
+                  className="card__bg"
+                  style={{
+                    backgroundImage: `url(${venue.mapBackgroundImageUrl})`,
+                    backgroundSize: "cover",
+                  }}
+                ></div>
+                <div className="card__info">
+                  <div
+                    className="card__logo"
+                    style={{ backgroundImage: `url(${venue.host?.icon})` }}
+                  ></div>
+                  <h3>{venue.name}</h3>
+                </div>
+                <Link className="card__button" to={`/admin_v2/${venue.id}`}>
+                  Manage Party
+                </Link>
+              </div>
+            );
+          })}
```

Comment:
We'll want to `useMemo` this `.map`.

We don't ever want to use `index` as a key as it's unstable/causes rendering issues. I believe it should be possible to use `venue.id` here?

It would also make sense to extract a `AdminVenueCard` or similar component that encapsulates this chunk here (not including the `.map` part)

Reply:
> We'll want to useMemo this .map

Why? This will rerender only if there is a change in `partyVenues` which will change only when `venues` changes. It will have no benefit only using resources for caching this.

Reply:
Sometimes we are overusing the memo/cache logic, I think this is one of those cases.

Reply:
> This will rerender only if there is a change in partyVenues which will change only when venues changes.

I don't believe that is correct.

This `.map()` will run every single time this component is re-rendered for any reason (which includes if it's own hooks/dependencies change, if any of its props change, or if literally any parent of it is re-rendered for any reason), even when `partyVenues` is exactly the same.

If you believe otherwise, I'd love for you to point to reference material that shows why that isn't the case; but this is my understanding of how React works, particular with regards to functional components as we use.

You could avoid a 'double useMemo' situation by just moving the `.map` into the existing useMemo, after the `.filter`, and then rendering that `const` directly in the JSX. The only logic here that seems to rely on it is the `.map` (that we would be moving into it anyway), and the `.length` (which would still work after the suggestion here)

Reply:
I understand what you are saying but in this case we have a single prop `venues` and nothing else. The parent doesn't have a state that can retrigger a render and this will always rely on `venues` If there is a change in `venues` the memo will trigger because of the dep array and the `.map` logic will go again.

Reply:
Are you saying that no parent/grandparent/etc component literally anywhere in the ‘parent chain’ between here and the very root of the app will ever render more than once under any circumstance? Because if not, then the issue I stated above remains true.

Unless the map is in the useMemo, it will run every time this component function is called. React can and does call component functions many times, even when ultimately it results in the same output. In those cases, it doesn’t modify the DOM, but it still had to run the entire component function to check that. This is a standard case of react component optimisation, and there is a lot written on it across the net.

An alternative would be to use React.memo on the entire component, at which point it may react how it sounds like you’re thinking it should react currently. If it’s purely driven by its props, has no internal state/connections/side effects that would change other than through those props, and contains no links to any child components that have any internal state/etc that would need to be triggered for any case other than these props changing, then that would be a fine approach. But if we do that, it entirely removes everything beneath this ‘node’ in the ‘render tree’ from being rendered, it’s a shortcut circuit. So if there was a child/grandchild/etc component somewhere in its hierarchy that was meant to listen/update based on redux state/similar, it wouldn’t anymore, which can lead to weird/hard to track down bugs. React.memo is a fairly nuclear option, and really only makes sense IMO where you can guarantee ‘pure’ ‘dumb components’ the whole way down, which IMO we rarely if ever can given the state of our platform code/design at the moment.

That’s why we tend to favour useMemo, and why I’m saying here (and other places where we map in the JSX) to do so.

It’s a simple matter of moving the .map inside the existing useMemo.

This is a blocker from me. If this gets merged without it, I will have to go and chase down the tech debt created and raise a PR to implement it. I don’t feel like this should be a contentious issue, and I’m kind of confused why you seem to be pushing back so strongly about it.

(cc // @mike-lvov perhaps you know of some good resources that can explain better than I have here? Or alternately that ‘disprove’ my claims here?)

If this is still an issue for you, please start a thread on slack, referring this comment thread, and let’s discuss.

Reply:
My intention is not to push this back and I'm sorry if it feels that I'm strongly trying to do so, I'm going to change it with my next commit, I don't want to force you to go back and clean my mess.

But like I already said we are overusing the memos and callbacks and there are many blogs stating that this should not be overused because sometimes the resources it takes to cache the data is more than it would take to re-render the component or run through the function again, usually in React you don't think about performance until you have a problem, I believe I saw this somewhere written by Dan Abramov or maybe in the React website itself. React is one of the fastest frameworks and you can re-render a component 10-15 times and the user won't notice because that's less than 50 milliseconds. I'm not staying stop using memo and callback, but maybe turn it down a notch and use it only when we have complex and heavy logic or big components that re-rendering them would slow don't the page. I believe the only place where this performance boost was hugely noticeable was when you refactored the grid to use callback and memo and the animation was smooth and pleasant because it was caching 5-10k div elements on the screen.